### ISSUE TITLE
>1000 tracks in friend scrobbler

### ISSUE THREAD

**Issue by theedgeofoblivious** [coczero/Last.fm-Scrubbler-WPF#83]
I decided that it would be better to use another name instead of the name I was using "last.fm/user/edgeofoblivious", and since there were issues exporting and importing with special characters(particularly accents in Spanish tracks), I checked out the features. The "Friend Scrobbler" feature works amazingly well. I can just scrobble from my old account to the new.

The one issue is that it only lets me select and import the first 1000 tracks. If I could choose more tracks(like up to the 2800 limit) and then if there was a way to select the subsequent 2800 and the subsequent 2800 and so on, that would be really useful. I have ~130,000 tracks I'd like to scrobble over to my new account. Thanks very much.

**Comment by coczero** [comment]
You could download a csv file of your first account then use the dsv scrobbler to scrobble them to account 2.
But you can only scrobble 2800 tracks per day.

**Comment by theedgeofoblivious** [comment]
I tried that, but there were significant problems. A lot of my music is in other languages which make use of accent marks and special characters, and whenever I would import those files, each character with an accent mark would come out as a group of multiple characters without accent marks. This meant that the track would appear in Last.fm, but it would appear as a new track with an unreadable incorrect name, and it wouldn't have the correct URL links on the Last.fm page and would appear without album artwork(with a generic icon).

I was aware of the 2800 limit, which with ~130,000 songs is already about 50 days worth of effort(using a bash script to split the 130,000-song CSV into 2,700-song chunks to allow an import of one such file per day along with scrobbling whatever I'd actually be listening to that day), but since I couldn't find a "replace all" function in the app and since so much of my music has accent marks, I found that it took roughly 45 minutes per day of extra effort to manually go through and fix each track name, artist name, and album name. After I'd done that for three days, I found that although I'd gotten it to correctly import the song names and artist names, none of the "Album Artist" names were identified correctly in Last.fm, and they were all identified by the mbdb instead, which explained why the songs were displaying no album artwork and their links weren't working. I had thought that was strange, because I had gone into the app's settings to choose the correct respective CSV columns for the respective scrobble tags, but I verified that that was the case.

I also discovered that the app had issues parsing. It seemed that commas and quotation marks within track titles, artist names, and album names often caused problems and made it so the app would report that it couldn't parse the CSV at all, so I used an app called CSV Lint to verify that my CSV files were all formatted as 100% proper CSV files. I also checked the line breaks to make sure they were using Windows format(CR+LF instead of just LF). I had to be really meticulous there, and I started to make sure that each file was 100% formatted properly. That took a combination of BBEdit (on Mac) and Notepad++(both of which have "find and replace all" functionality) and LibreOffice, and more time. And even when the CSV files were 100% correct, they wouldn't necessarily be recognized by the app. It seemed that the app would import them in one mode but not the other. I spent about three hours for each of the three nights in a row(Windows 10 with all updates and version 1.21 beta of the app, and I also tried editing the files on BBEdit on the Mac, and CSV Lint verified that the files it produced were Windows-formatted CSV files with zero errors). Finding out that almost none of the albums were identified correctly and that I'd have to completely remove the scrobble history from my new account and start again after spending about ten hours and three days of time made me feel like it might not be worth continuing(because I was estimating three hours times ~50 days, or about 150 hours of work). I had to use TamperMonkey and a JQuery javascript to automate the clicking of each of the thousands of "Remove track" links on my last.fm profile.

The app's "Friend Scrobbler" feature was very limited in terms of the number of songs it would import. It seems to have only successfully imported 250 or so, but all of the tags were correct, it only took a few seconds, and it required very little effort. I was impressed with that, and if it was capable of going through and importing subsequent blocks of up to 2800 and going through the "friend" account's whole library then that would make it possible to import the whole library.
